internal/k8s/epslices/endpoint_slices.go

```diff
@@ -11,16 +11,11 @@ import (
 
 const SlicesServiceIndexName = "ServiceName"
 
-// IsConditionServing tells if the conditions represent a serving state, deferring
-// to ready state if serving == nil.
-func IsConditionServing(conditions discovery.EndpointConditions) bool {
-	if conditions.Serving == nil {
-		if conditions.Ready == nil {
-			return true
-		}
-		return *conditions.Ready
-	}
-	return *conditions.Serving
+// IsConditionReadyOrServing tells if the conditions represent a ready state or serving state is ready.
+func IsConditionReadyOrServing(conditions discovery.EndpointConditions) bool {
```

Comment:
I think this can be untangled a bit more, with something like:

```go
/ IsConditionReadyOrServing tells if the conditions represent a ready state or serving state is ready.
func IsConditionReadyOrServing(conditions discovery.EndpointConditions) bool {
        if conditions.Ready == nil || *conditions.Ready {
                return true
        }
        if conditions.Serving != nil && *conditions.Serving {
                return true
        }
        return false
}
```

I'd also rename to endpointCanServe as it is the semantic we are looking for